### PR TITLE
ADVI test, random seed. 

### PR DESCRIPTION
**Pull request by taku-y**
Add tests to check the program runs without error for a model. The seed to initialize random state can be set. Fix the code for checking discrete random variables.

### CODE REVIEW

File: pymc3/tests/test_advi.py [hunk 1]
```diff
@@ -90,4 +90,42 @@ def create_minibatch():
...
+
+def test_advi():
+    data = np.random.randn(100)
+
+    with Model() as model: 
+        mu = Normal('mu', mu=0, sd=1, testval=0)
+        sd = HalfNormal('sd', sd=1)
+        n = Normal('n', mu=mu, sd=sd, observed=data)
+
+    means, sds, elbos = advi(model=model, n=10, accurate_elbo=True)
```
Comment (twiecki) [inline pymc3/tests/test_advi.py]: Could add an easy test to make sure the correct values are retrieved.

File: pymc3/tests/test_advi.py [hunk 2]
```diff
@@ -90,4 +90,42 @@ def create_minibatch():
...
+            data = np.roll(data, 100, axis=0)
+            yield data[:100]
+
+    minibatches = [create_minibatch(data)]
+
+    means, sds, elbos = advi_minibatch(
+        model=model, n=10, minibatch_tensors=minibatch_tensors, 
+        minibatch_RVs=minibatch_RVs, minibatches=minibatches, 
+        total_size=total_size
+    )
```
Comment (twiecki) [inline pymc3/tests/test_advi.py:154]: Same.

File: pymc3/variational/advi.py [hunk 3]
```diff
@@ -222,7 +216,7 @@ def run_adagrad(uw, grad, elbo, n, learning_rate=.001, epsilon=.1, verbose=1):
 
 def variational_gradient_estimate(
     vars, model, minibatch_RVs=[], minibatch_tensors=[], total_size=None, 
-    n_mcsamples=1):
+    n_mcsamples=1, seed=1):
```
Comment (twiecki) [inline pymc3/variational/advi.py]: Should we default to `None` here so that the seed is not set by default?

File: pymc3/tests/test_advi.py (hunk 1 above)
Comment (taku-y) [inline pymc3/tests/test_advi.py]: Add a check on the inference of the posterior mean of Gaussian distribution.

Comment (twiecki) [inline pymc3/tests/test_advi.py]: Are you asking or making a note to yourself?

File: pymc3/variational/advi.py (hunk 3 above)
Comment (taku-y) [inline pymc3/variational/advi.py]: As in scikit-learn, if None is given, seed is initialized based on current state of the machine. The random number generator defined in Theano does not accept None as an argument, so add lines to make the seed random.

Comment (twiecki) [inline pymc3/variational/advi.py]: Interesting, you're correct. The default initialize is to `12345` (http://deeplearning.net/software/theano/library/sandbox/rng_mrg.html). In that case, we shouldn't randomize the seed but just use the same default.

File: pymc3/tests/test_advi.py (hunk 1 above)
Comment (taku-y) [inline pymc3/tests/test_advi.py]: It's just a note what I did. Sorry for not making clear.

File: pymc3/variational/advi.py (hunk 3 above)
Comment (taku-y) [inline pymc3/variational/advi.py]: Thanks for your comment. I'll fix it soon.

**Comment by twiecki** [comment]
Thanks @taku-y!
